### PR TITLE
added ability to set custom template delims

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Description of your changes
Added ability to set custom template delims to avoid using escaping, while using Helm

### CODE REVIEW

File: input/v1beta1/input.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ type GoTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// Template start characters
+	LeftDelims string `json:"leftDelims,omitempty"`
+	// Template end characters
+	RightDelims string `json:"rightDelims,omitempty"`
```
Comment (negz) [inline input/v1beta1/input.go:25]:
```suggestion
	// Template start characters
	// +optional
	LeftDelims *string `json:"leftDelims,omitempty"`
	// Template end characters
	// +optional
	RightDelims *string `json:"rightDelims,omitempty"`
	
```

These are optional, right? Even though this isn't a real CRD we try to follow CRD conventions. So for optional fields we want them to be pointers, and have the `+optional` comment marker.

**Comment by negz** [comment]
Could you please add a new directory under `example` showing how to use these? The example should have everything it needs to be run with `crossplane beta render`. Thank you!

File: example/custom-delims/composition.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  mode: Pipeline
+  pipeline:
+    - step: render-templates
+      functionRef:
+        name: function-go-templating
+      input:
+        apiVersion: gotemplating.fn.crossplane.io/v1beta1
+        kind: GoTemplate
+        leftDelims: '[['
+        rightDelims: ']]'
```
Comment (turkenh) [inline example/custom-delims/composition.yaml:18]:
```suggestion
      delims:
        left: '[['
        right: ']]'
```

or should we even nest under `config` or something if we envision more configuration in the future?

```
config:
  delims:
    left: '[['
    right: ']]'
```

**Comment by ezgidemirel** [comment]
Hi @AndrewChubatiuk, thanks for your PR! It looks like some of the unit tests are failing. Could you please fix them and sign your commits?

File: input/v1beta1/input.go [hunk 3]
```diff
@@ -18,7 +18,9 @@ import (
 type GoTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
+	// Go Template Config
+	// +optional
+	Config *Config `json:"config,omitempty"`
```
Comment (phisco) [inline input/v1beta1/input.go:23]: maybe config is a bit generic? all the other fields below (source/inline/filesystem) could easily be below `config`. WDYT about `templateConfig`? CC @turkenh

**Review by phisco — APPROVED** [review]
LGTM! thanks @AndrewChubatiuk!

**Review by ezgidemirel — APPROVED** [review]
Thanks @AndrewChubatiuk!

**Pull request merged**